src/runtime/fake_ion.cpp

```diff
@@ -2,19 +2,39 @@
 
 namespace Halide { namespace Runtime { namespace Internal { namespace Ion {
 
+struct allocation_record {
+    void *original;
+};
+
 // Allocate an ION buffer, and map it, returning the mapped pointer.
 WEAK void *ion_alloc(void *user_context, size_t len, int heap_id, int *out_fd) {
-    void *ret = halide_malloc(user_context, len);
+    const size_t align = 128;
+
+    // Align the allocation size.
+    len = (len + align - 1) & ~(align - 1);
+
+    // Allocate enough space to hold information about the allocation prior to the pointer we return.
+    len += align;
+
+    void *original = halide_malloc(user_context, len);
+
+    // Store the original ptr before the pointer we return.
+    allocation_record rec = { original };
+    void *ret = reinterpret_cast<char *>(original) + align;
```

Comment:
Sorry, yeah, I noticed right after I commented that it was closed, I spent too long staring at it trying to find the alignment :)